galata/test/jupyterlab/settings.test.ts

```diff
@@ -31,9 +31,25 @@ test('Open the settings editor with a specific search query', async ({
 
   const settingsPanel = page.locator('.jp-SettingsPanel');
 
-  expect(await settingsPanel.screenshot()).toMatchSnapshot(
-    'settings-panel.png'
-  );
+  expect
+    .soft(await settingsPanel.screenshot())
+    .toMatchSnapshot('settings-panel.png');
+  // Test that new query takes effect
+  await expect(page.locator('.jp-PluginList-entry')).toHaveCount(1);
+
+  await page.evaluate(async () => {
+    await window.jupyterapp.commands.execute('settingeditor:open', {
+      query: 'CodeMirror'
+    });
+  });
+  // wait for command to be executed
+  await page.waitForTimeout(1000);
+  const listEntry = await page
+    .locator('.jp-PluginList-entry-label-text')
+    .first();
+  const textContent = await listEntry.textContent();
+
+  expect(textContent).toBe('CodeMirror');
```

Comment:
Let's avoid  explicit timeouts as these are easy to add but accumulate quickly and slow down the test suite. Can you try:

```suggestion
  // wait for command to be executed
  const fistListEntry = page
    .locator('.jp-PluginList-entry-label-text')
    .first();

  await expect(fistListEntry).toHaveText('CodeMirror');
```

`toHaveText` is an auto-waiting assertion and by default waits for up to 5 seconds.

Reply:
Awesome! I tried a couple of things to avoid having the `waitForTimeout` call, but none of them worked. I just tried this out locally and it passes.